src_Core/RISCY_OOO/procs/lib/Decode.bsv

```diff
@@ -502,18 +508,31 @@ function DecodeResult decode(Instruction inst, Bool cap_mode);
         end
 
         opcJalr: begin
-            dInst.iType = Jr;
             regs.dst  = Valid(tagged Gpr rd);
             regs.src1 = Valid(tagged Gpr rs1);
             regs.src2 = Invalid;
             dInst.imm = Valid(immI);
             dInst.csr = tagged Invalid;
             dInst.execFunc = tagged Br AT;
+
             dInst.capChecks.check_enable = True;
-            dInst.capChecks.check_authority_src = Pcc;
             dInst.capChecks.check_low_src = Src1Addr;
             dInst.capChecks.check_high_src = Src1AddrPlus2;
             dInst.capChecks.check_inclusive = True;
+            if (cap_mode) begin
+                dInst.iType = CJALR;
+
+                dInst.capChecks.src1_tag = True;
+                dInst.capChecks.src1_permit_x = True;
+                dInst.capChecks.src1_unsealed_or_sentry = True;
```

Comment:
Need to be careful; JALR has an immediate and we don't want that to allow you to modify sentries (ie sentries are only allowed if the immediate is 0)

Reply:
Good point! I guess that's a seal_exception on the source, i.e. the same as trying to JALR to a sealed_with_type cap?

Reply:
That's what I intended to do in QEMU :)